apps/src/templates/progress/progressHelpers.js

```diff
@@ -118,6 +121,54 @@ export function stageIsAllAssessment(levels) {
   return levels.every(level => level.kind === LevelKind.assessment);
 }
 
+/**
+ * Computes progress status percentages for a set of levels.
+ * @param {{id:studentLevelProgressType}} studentProgress An object keyed by
+ * level id containing objects representing the student's progress in that level
+ * @param {levelType[]} levels An array of levels
+ * @returns {object} An object with a percentage of levels in each of the
+ * following buckets: attempted, imperfect, completed, incomplete.
+ *
+ * Note: this function will replace `summarizeProgressInStage` below once we
+ * refactor the legacy StudentProgressSummaryCell component
+ */
+export function statusPercentsForLevels(studentProgress, levels) {
+  // Filter any bonus levels as they do not count toward progress.
+  const filteredLevels = levels.filter(level => !level.bonus);
+  const statuses = filteredLevels.map(level => {
+    const levelProgress = studentProgress[level.id];
+    return (levelProgress && levelProgress.status) || LevelStatus.not_tried;
```

Comment:
whaaa? i did not know js had an optional chaining operator!! but yeah would need some sort of polyfill since it's not supported in all browsers. i would vote for making the decision to start incorporating that into our codebase with the broader team rather than in this PR. but i'm all for it!